airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -83,4 +132,347 @@ protected DataType<?> toDialectType(final AirbyteProtocolType airbyteProtocolTyp
+    }
+    return condition;
+  }
+
+  @Override
+  public String createTable(final StreamConfig stream, final String suffix, final boolean force) {
+    // TODO: Use Naming transformer to sanitize these strings with redshift restrictions.
+    String finalTableIdentifier = stream.id().finalName() + suffix.toLowerCase();
+    if (!force) {
+      return Strings.join(
+          List.of(
+              createSchemaSql(stream.id().finalNamespace()),
+              createTableSql(stream.id().finalNamespace(), finalTableIdentifier, stream.columns())),
+          ";" + System.lineSeparator());
+    }
+    return Strings.join(
+        List.of(
+            createSchemaSql(stream.id().finalNamespace()),
+            beginTransaction(),
+            dropTableIfExists(quotedName(stream.id().finalNamespace(), finalTableIdentifier)),
+            createTableSql(stream.id().finalNamespace(), finalTableIdentifier, stream.columns()),
+            commitTransactionInternal()),
+        ";" + System.lineSeparator());
+  }
+
+  @Override
+  public String updateTable(final StreamConfig streamConfig,
+                            final String finalSuffix,
+                            final Optional<Instant> minRawTimestamp,
+                            final boolean useExpensiveSaferCasting) {
+
+    // TODO: Add flag to use merge vs insert/delete
+    return insertAndDeleteTransaction(streamConfig, finalSuffix, minRawTimestamp, useExpensiveSaferCasting);
+
+  }
+
+  @Override
+  public String overwriteFinalTable(final StreamId stream, final String finalSuffix) {
+    return Strings.join(
+        List.of(
+            dropTableIfExists(name(stream.finalNamespace(), stream.finalName())),
+            alterTable(name(stream.finalNamespace(), stream.finalName() + finalSuffix))
+                .renameTo(name(stream.finalName()))
+                .getSQL()),
+        ";" + System.lineSeparator());
+  }
+
+  @Override
+  public String migrateFromV1toV2(final StreamId streamId, final String namespace, final String tableName) {
+    final Name rawTableName = name(streamId.rawNamespace(), streamId.rawName());
+    return Strings.join(
+        List.of(
+            createSchemaIfNotExists(streamId.rawNamespace()).getSQL(),
+            dropTableIfExists(rawTableName).getSQL(),
+            DSL.createTable(rawTableName)
+                .column(COLUMN_NAME_AB_RAW_ID, SQLDataType.VARCHAR(36).nullable(false))
+                .column(COLUMN_NAME_AB_EXTRACTED_AT, SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false))
+                .column(COLUMN_NAME_AB_LOADED_AT, SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false))
+                .column(COLUMN_NAME_DATA, getStructType().nullable(false))
+                .as(select(
+                    field(COLUMN_NAME_AB_ID).as(COLUMN_NAME_AB_RAW_ID),
+                    field(COLUMN_NAME_EMITTED_AT).as(COLUMN_NAME_AB_EXTRACTED_AT),
+                    inline(null, SQLDataType.TIMESTAMPWITHTIMEZONE).as(COLUMN_NAME_AB_LOADED_AT),
```

Comment:
https://github.com/airbytehq/airbyte/pull/33367 got merged! so you can rebase in those changes now